app/models/shf_application.rb

```diff
@@ -195,6 +198,18 @@ def add_observers
   end
 
 
+  # @return [boolean] - is the application in a state where the applicant can edit it?
+  def applicant_can_edit?
+    CAN_EDIT_STATES.include? state.to_sym
+  end
+
+
+  # @return [Boolean] - can the applicant upload files for this application?
+  def applicant_can_upload_files?
+    applicant_can_edit? || (accepted? && user.membership_app_and_payments_current?)
+  end
```

Comment:
Since this PR has to do with renewals, and renewals require that a file be uploaded and attached to the _user_, why is this here?

Reply:
Leftover.   Should have been deleted.  Good catch.